src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptionsUiCsvLocale.java

```diff
@@ -123,10 +131,46 @@ private void initializeDefaults()
       encoding_.addItem("UTF-8-MAC", "UTF-8-MAC");
       encoding_.addItem("UTF8", "UTF8");
       encoding_.addItem("UTF8-MAC", "UTF8-MAC");
+      encoding_.addItem("Other", "");
       
       encoding_.setSelectedIndex(8);
    }
+   
+   private void replaceOtherEncodingItem(int selectedIndex, String otherEncoding) {
+      encoding_.removeItem(selectedIndex);
+      encoding_.insertItem("Other", otherEncoding, selectedIndex);
+      encoding_.setSelectedIndex(selectedIndex);
+   }
 
+   private void initializeEvents()
+   {
+      ChangeHandler encodingChangeHandler = new ChangeHandler()
+      {
+         @Override
+         public void onChange(ChangeEvent arg0)
+         {
+            if (encoding_.getSelectedItemText() == "Other")
+            {
+               globalDisplay_.promptForText(
+                  "Encoding Identifier",
+                  "Please enter an encoding identifier. For a list of valid encodings run iconvlist().",
+                  encoding_.getSelectedValue(),
+                  new OperationWithInput<String>()
+                  {
+                     @Override
+                     public void execute(final String otherEncoding)
+                     {
+                        int selectedIndex = encoding_.getSelectedIndex();
```

Comment:
This dialog already prevents users from entering empty strings so they will have to hit "cancel" in this case.